src/ring_theory/integral_closure.lean

```diff
@@ -27,20 +29,37 @@ open_locale big_operators
 open polynomial submodule
 
 section ring
-variables (R : Type*) {A : Type*}
+variables {R A : Type*}
 variables [comm_ring R] [ring A]
-variables [algebra R A]
+
+/-- An element `x` of `A` is said to be integral over `R` with respect to `f`
+if it is a root of a monic polynomial `p : polynomial R` evaluated under `f` -/
+def ring_hom.is_integral_elem (f : R →+* A) (x : A) :=
+∃ p : polynomial R, monic p ∧ eval₂ f x p = 0
+
+/-- A ring homomorphism `f : R →+* A` is said to be integral
+if every element `A` is integral with respect to the map `f` -/
+def ring_hom.is_integral (f : R →+* A) :=
+∀ x : A, f.is_integral_elem x
+
+variables [algebra R A] (R)
 
 /-- An element `x` of an algebra `A` over a commutative ring `R` is said to be *integral*,
-if it is a root of some monic polynomial `p : polynomial R`. -/
+if it is a root of some monic polynomial `p : polynomial R`.
+Equivalently, the element is integral over `R` with respect to the induced `algebra_map` -/
 def is_integral (x : A) : Prop :=
-∃ p : polynomial R, monic p ∧ aeval x p = 0
+(algebra_map R A).is_integral_elem x
```

Comment:
It's a bit confusing that the split is made between `ring_hom.is_integral_elem` and `ring_hom.is_integral` one way, and between `_root_.is_integral` and `algebra.is_integral` another way. On the other hand, this way more closely matches existing mathematical terminology. I can't think of any better alternatives, so I guess let's keep it as is for now.